Cannot install xmlrpc on Mac BigSur with python 3.8.6

```
✗ pkg-config --cflags xmlsec1
-D__XMLSEC_FUNCTION__=__func__ -DXMLSEC_NO_SIZE_T -DXMLSEC_NO_GOST=1 -DXMLSEC_NO_GOST2012=1 -DXMLSEC_NO_CRYPTO_DYNAMIC_LOADING=1 -DXMLSEC_CRYPTO_OPENSSL=1 -I/usr/local/Cellar/libxml2/2.9.10_2/include/libxml2 -I/usr/local/Cellar/libxmlsec1/1.2.32/include/xmlsec1 -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/openssl@1.1/include/openssl
```

```
 ✗ pip install xmlsec
Collecting xmlsec
  Using cached xmlsec-1.3.9.tar.gz (62 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
Requirement already satisfied: lxml>=3.8 in /Users/slysenko/.pyenv/versions/3.8.6/lib/python3.8/site-packages (from xmlsec) (4.6.3)
Building wheels for collected packages: xmlsec
  Building wheel for xmlsec (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/slysenko/.pyenv/versions/3.8.6/bin/python3.8 /Users/slysenko/.pyenv/versions/3.8.6/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/tmp72ud_82q
       cwd: /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34
  Complete output (51 lines):
  running bdist_wheel
  running build
  running build_py
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/overlay/lib/python3.8/site-packages/setuptools/dist.py:642: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
    warnings.warn(
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/overlay/lib/python3.8/site-packages/setuptools/dist.py:642: UserWarning: Usage of dash-separated 'build-requires' will not be supported in future versions. Please use the underscore name 'build_requires' instead
    warnings.warn(
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/overlay/lib/python3.8/site-packages/setuptools/dist.py:642: UserWarning: Usage of dash-separated 'upload-dir' will not be supported in future versions. Please use the underscore name 'upload_dir' instead
    warnings.warn(
  package init file 'src/xmlsec/__init__.py' not found (or not a regular file)
  creating build
  creating build/lib.macosx-11.2-x86_64-3.8
  creating build/lib.macosx-11.2-x86_64-3.8/xmlsec
  copying src/xmlsec/py.typed -> build/lib.macosx-11.2-x86_64-3.8/xmlsec
  copying src/xmlsec/tree.pyi -> build/lib.macosx-11.2-x86_64-3.8/xmlsec
  copying src/xmlsec/__init__.pyi -> build/lib.macosx-11.2-x86_64-3.8/xmlsec
  copying src/xmlsec/constants.pyi -> build/lib.macosx-11.2-x86_64-3.8/xmlsec
  copying src/xmlsec/template.pyi -> build/lib.macosx-11.2-x86_64-3.8/xmlsec
  running build_ext
  building 'xmlsec' extension
  creating build/temp.macosx-11.2-x86_64-3.8
  creating build/temp.macosx-11.2-x86_64-3.8/private
  creating build/temp.macosx-11.2-x86_64-3.8/private/var
  creating build/temp.macosx-11.2-x86_64-3.8/private/var/folders
  creating build/temp.macosx-11.2-x86_64-3.8/private/var/folders/yr
  creating build/temp.macosx-11.2-x86_64-3.8/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp
  creating build/temp.macosx-11.2-x86_64-3.8/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T
  creating build/temp.macosx-11.2-x86_64-3.8/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i
  creating build/temp.macosx-11.2-x86_64-3.8/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34
  creating build/temp.macosx-11.2-x86_64-3.8/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -I/usr/local/opt/libxml2/include -D__XMLSEC_FUNCTION__=__func__ -DXMLSEC_NO_SIZE_T -DXMLSEC_NO_GOST=1 -DXMLSEC_NO_GOST2012=1 -DXMLSEC_NO_CRYPTO_DYNAMIC_LOADING=1 -DXMLSEC_CRYPTO_OPENSSL=1 -DMODULE_NAME=xmlsec -DMODULE_VERSION=1.3.9 -I/usr/local/Cellar/libxml2/2.9.10_2/include/libxml2 -I/usr/local/Cellar/libxmlsec1/1.2.32/include/xmlsec1 -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/openssl@1.1/include/openssl -I/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/normal/lib/python3.8/site-packages/lxml/includes -I/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/normal/lib/python3.8/site-packages/lxml -I/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/normal/lib/python3.8/site-packages/lxml/includes/libxml -I/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/normal/lib/python3.8/site-packages/lxml/includes/libxslt -I/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/normal/lib/python3.8/site-packages/lxml/includes/libexslt -I/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-build-env-9voggw_z/normal/lib/python3.8/site-packages/lxml/includes/__pycache__ -I/Users/slysenko/.pyenv/versions/3.8.6/include/python3.8 -c /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src/constants.c -o build/temp.macosx-11.2-x86_64-3.8/private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src/constants.o -g -std=c99 -fPIC -fno-strict-aliasing -Wno-error=declaration-after-statement -Werror=implicit-function-declaration -Os
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src/constants.c:321:5: error: use of undeclared identifier 'xmlSecXkmsNs'
      PYXMLSEC_ADD_NS_CONSTANT(XkmsNs, "XKMS");
      ^
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src/constants.c:311:45: note: expanded from macro 'PYXMLSEC_ADD_NS_CONSTANT'
      tmp = PyString_FromString((const char*)(JOIN(xmlSec, name))); \
                                              ^
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src/common.h:19:19: note: expanded from macro 'JOIN'
  #define JOIN(X,Y) DO_JOIN1(X,Y)
                    ^
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src/common.h:20:23: note: expanded from macro 'DO_JOIN1'
  #define DO_JOIN1(X,Y) DO_JOIN2(X,Y)
                        ^
  /private/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/pip-install-eocity2i/xmlsec_6b9b499ff62c45b3871ae6c495b2cc34/src/common.h:21:23: note: expanded from macro 'DO_JOIN2'
  #define DO_JOIN2(X,Y) X##Y
                        ^
  <scratch space>:12:1: note: expanded from here
  xmlSecXkmsNs
  ^
  1 error generated.
  error: command 'clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for xmlsec
Failed to build xmlsec
ERROR: Could not build wheels for xmlsec which use PEP 517 and cannot be installed directly
```


libxml2-2.9.10_2
libxmlsec1--1.2.32
pkg-config--0.29.2_3

--- a/src/constants.c
+++ b/src/constants.c
@@ -452,7 +452,8 @@ int PyXmlSec_ConstantsModule_Init(PyObject* package) {
 #ifndef XMLSEC_NO_DSA
     PYXMLSEC_ADD_KEYDATA_CONSTANT(KeyDataDsa, "DSA")
 #endif
-#if XMLSEC_VERSION_HEX > 306
+#if XMLSEC_VERSION_HEX > 0x10212
+    // from version 1.2.19
     PYXMLSEC_ADD_KEYDATA_CONSTANT(KeyDataEcdsa, "ECDSA")
 #endif
     PYXMLSEC_ADD_KEYDATA_CONSTANT(KeyDataHmac, "HMAC")
@@ -502,7 +503,8 @@ int PyXmlSec_ConstantsModule_Init(PyObject* package) {
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformDsaSha1, "DSA_SHA1");
 #endif
 
-#if XMLSEC_VERSION_HEX > 306
+#if XMLSEC_VERSION_HEX > 0x10212
+    // from version 1.2.19
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformEcdsaSha1, "ECDSA_SHA1");
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformEcdsaSha224, "ECDSA_SHA224");
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformEcdsaSha256, "ECDSA_SHA256");
@@ -543,7 +545,8 @@ int PyXmlSec_ConstantsModule_Init(PyObject* package) {
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformSha384, "SHA384");
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformSha512, "SHA512");
 
-#if XMLSEC_VERSION_HEX > 315
+#if XMLSEC_VERSION_HEX > 0x1021B
+    // from version 1.2.28
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformAes128Gcm, "AES128_GCM");
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformAes192Gcm, "AES192_GCM");
     PYXMLSEC_ADD_TRANSFORM_CONSTANT(TransformAes256Gcm, "AES256_GCM");

--- a/src/main.c
+++ b/src/main.c
@@ -27,7 +27,7 @@ static int free_mode = _PYXMLSEC_FREE_NONE;
 
 #ifndef XMLSEC_NO_CRYPTO_DYNAMIC_LOADING
 static const xmlChar* PyXmlSec_GetCryptoLibName() {
-#if XMLSEC_VERSION_HEX > 308
+#if XMLSEC_VERSION_HEX > 0x10214
     // xmlSecGetDefaultCrypto was introduced in version 1.2.21
     const xmlChar* cryptoLib = xmlSecGetDefaultCrypto();
 #else

--- a/src/platform.h
+++ b/src/platform.h
@@ -19,11 +19,11 @@
 #include <windows.h>
 #endif /* MS_WIN32 */
 
-#define XMLSEC_VERSION_HEX ((XMLSEC_VERSION_MAJOR << 8) | (XMLSEC_VERSION_MINOR << 4) | (XMLSEC_VERSION_SUBMINOR))
+#define XMLSEC_VERSION_HEX ((XMLSEC_VERSION_MAJOR << 16) | (XMLSEC_VERSION_MINOR << 8) | (XMLSEC_VERSION_SUBMINOR))
 
 // XKMS support was removed in version 1.2.21
 // https://mail.gnome.org/archives/commits-list/2015-February/msg10555.html
-#if  XMLSEC_VERSION_HEX > 0x134
+#if  XMLSEC_VERSION_HEX > 0x10214
 #define XMLSEC_NO_XKMS 1
 #endif
 

